fedimint-server/src/net/api.rs

```diff
@@ -296,6 +310,73 @@ impl ConsensusApi {
             .await
             .map_err(|_| ApiError::server_error("Unable send event".to_string()))
     }
+
+    pub async fn build_full_server_status(&self) -> ApiResult<FullStatus> {
+        let our_epoch = self.get_epoch_count().await;
+        // Clone to avoid holding the lock while we calculate things
+        let rejoin_at_epoch = self.rejoin_at_epoch.lock().await.clone();
+        let last_epoch_by_peer = rejoin_at_epoch.into_iter().fold(
+            HashMap::<PeerId, u64>::new(),
+            |mut acc, (epoch, peers)| {
+                let epoch = epoch + 1;
+                for peer in peers {
+                    match acc.entry(peer) {
+                        std::collections::hash_map::Entry::Occupied(mut entry) => {
+                            // Only update if the epoch is higher
+                            if epoch > *entry.get() {
+                                entry.insert(epoch);
+                            }
+                        }
+                        std::collections::hash_map::Entry::Vacant(entry) => {
+                            entry.insert(epoch);
+                        }
+                    }
+                }
+                acc
+            },
+        );
+        let peers_status = self.peer_status_client.query_all_status().await;
```

Comment:
Eehh... so we went with the active queering approach? :/

Reply:
Yeah, I thought this would be controversial :smile: 

It was my first idea and It was very similar to one of @elsirion's ideas, so I left it as is.

The only sort of "passive" approach I can think of is to create a "events" channel on `CommonPeerConnectionState` and write to it after each state transition and let a thread accumulate this state. Then we would query this state instead of the original one.

We can try that.

Reply:
Actually, nvmd. This is a a call to send a request to peer threads. I wish we could use some other names than "query" and "client". 

IMO this should be

```rust
self.peer_status.get();
```

or something. The fact that there are some "rpc"-like things there can be an implementation detail.